deegree-core/deegree-core-base/src/main/java/org/deegree/filter/utils/FilterUtils.java

```diff
@@ -58,6 +61,14 @@ public static Geometry getGeometryValue( TypedObjectNode node ) {
             geom = (Geometry) node;
         } else if ( node instanceof Property && ( (Property) node ).getValue() instanceof Geometry ) {
             geom = (Geometry) ( (Property) node ).getValue();
+        } else if ( node instanceof GenericXMLElement ) {
+            GenericXMLElement xml = (GenericXMLElement) node;
+            if ( xml.getChildren() != null && !xml.getChildren().isEmpty() ) {
```

Comment:
If GenericXMLElement is a collection type, it would be preferable that the size of the collection is also checked for exactly one element.